Fix a bunch of Galaxy tests

mvdbeek opened the pull request:
And coincidentally makes the green required conformance tests green again

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

mvdbeek committed:
Ignore fals-y scatter_type

mvdbeek committed:
Add fields parameter to create collection schema

mvdbeek committed:
Fix non_data_connection workflows

mvdbeek committed:
Use staged_addition/add_pending_items

mvdbeek committed:
Use allow_tool_state_corrections when running CWL conformance tests

mvdbeek committed:
Can't used TypedDict in python < 3.8

mvdbeek committed:
Fix inverted worklfow ids and invocation ids

mvdbeek committed:
Fix nested collections getting reduced completely

I think that might have just not been merged properly when we fixed this
in upstream Galaxy (https://github.com/galaxyproject/galaxy/commit/4df1de38ba390ee6513bc85fbb0643e6e152bd25).

mvdbeek committed:
Fix handling of uploaded_file_name

mvdbeek committed:
Reenable check_upload_content

mvdbeek commented:
So API + Selenium should be passing completely now.

nsoranzo commented on lib/galaxy_test/driver/driver_util.py:
```diff
@@ -229,7 +229,7 @@ def setup_galaxy_config(
         auto_configure_logging=logging_config_file is None,
         check_migrate_tools=False,
-        check_upload_content=False,
+        check_upload_content=True,
```
Since that's the default, we can just remove the line (and reduce the diff ;)
```suggestion
```

mvdbeek committed:
Undo persisting ephemeral collections

So revert https://github.com/common-workflow-language/galaxy/pull/128
bascially.

nsoranzo committed:
Remove unneeded `check_upload_content=True`

mvdbeek committed:
Fix directory location tests

mvdbeek committed:
Consider a history not new once there are any items in it

Fixes waiting for history state to become OK if the history contains
only empty collections.

nsoranzo commented on lib/galaxy/managers/histories.py:
```diff
@@ -527,7 +527,7 @@ def serialize_history_state(self, item, key, **context):
         #   so this is rel. ineff. - if we keep this...
         hda_state_counts = self.serialize_state_counts(history, 'counts', exclude_deleted=True, **context)
         num_hdas = sum(hda_state_counts.values())
```
Can you move this to the `else` branch?

mvdbeek committed:
Move num_hdas to else branch
